app/src/components/market/sections/market_list/market_home_container.tsx

```diff
@@ -166,7 +244,44 @@ const MarketHomeContainer: React.FC = () => {
     setFilter(filter)
     setMoreMarkets(true)
     setIsFiltering(true)
-  }, [])
+
+    let route: string = ''
+    let routeQuery: string = '?'
+    
+    if(filter.sortBy === 'lastActiveDayAndRunningDailyVolume') {
+      route += '/24h-volume'
+    } else if(filter.sortBy === 'collateralVolume') {
+      route += '/volume'
+    } else if(filter.sortBy === 'creationTimestamp') {
+      route += '/newest'
+    } else if(filter.sortBy === 'openingTimestamp') {
+      route += '/ending'
+    } else if(filter.sortBy === 'liquidityParameter') {
+      route += '/liquidity'
+    }
+
+    if(filter.currency) {
+      route += `/currency/${filter.currency}`
+    }
+
+    if(filter.arbitrator) {
+      route += `/arbitrator/${filter.arbitrator}`
+    }
+
+    if(filter.category && filter.category !== 'All') {
+      route += `/category/${filter.category}`
+    }
+
+    if(filter.state && filter.state !== 'OPEN') {
+      routeQuery += `state=${filter.state}&`
+    }
+
+    if(filter.title) {
+      routeQuery += `tag=${filter.title}&`
```

Comment:
Maybe routeQuery could be an array first with each query, and then do routeQuery.join('&').